Entity/Player/Player.gd

```diff
@@ -2,11 +2,50 @@ extends Entity
 class_name Player
 ## Base player Entity class.
 
+var _should_save_persistent_data: bool = true
+
 # @Override
 func _ready() -> void:
 	super()
 	PlayerManager.set_player(self)
+	
+	_load_persistent_data()
 
 
 func _exit_tree() -> void:
+	# Save player data when they are destroyed (combat end)
 	PlayerManager.set_player(null)
+	
+	if _should_save_persistent_data:
+		_save_persistent_data()
+
+
+# Set values from our saved persistent data
+func _load_persistent_data() -> void:
```

Comment:
we might need to set other stuff later (like objects for example, though that might be handled separately of the player, in the same way that cards and deck is handled separately

Reply:
What do you mean by objects? Like player's inventory?

Reply:
yes, items
not really anything to change based on my comment, i was just commenting on that